### ISSUE TITLE
Can't remove last tag on an order

### ISSUE THREAD

**Issue by wassimk** [tradegecko/gecko#64]
Hey,

I have added an attribute to order called tags and set it as an array. I'm able to add new tags and even remove tags by updating the array and saving the order object. But, when it comes down to the last tag on the order I can't remove it. I set tags to an empty array and save but the last tag remains. I even tried setting it to nil, same problem.

I recorded a video showing what I'm doing. Can you take a look and let me know what I'm missing? 
https://shipzen.wistia.com/medias/rnl43zjpwm

The use case: I have a feature in another app that reserves orders by tagging them, those orders don't get shipped out via our 3PL process. When the unreserve date arrives the app automatically unreserves the order. Part of the unreserve process is removing the reserved tag which then lets the order ship out. I can no longer remove the [last | only] tag.

**Comment by lcpriest** [comment]
@wassimk what happens if you send an empty string?

**Comment by wassimk** [comment]
Hey @lcpriest , looks like it's smart and sets it to an array with one item which is an empty string.

``` ruby
> o.tags
=> []
> o.tags = ''
=> ""
> o.save
=> true
> o.tags
=> [""]
```

**Comment by lcpriest** [comment]
If you reload that (fetch it again), is it still an empty string? I can't see any use case for it working this way.

**Comment by lcpriest** [comment]
What is the `o.as_json`? It might be stripping non present values from the payload.

**Comment by wassimk** [comment]
Yes, if I reload it it comes back as an array with the single empty string. In actually shows up as an empty string in the UI.

The `o.as_json` returns exactly what we set. If empty string it will give `['']` if empty array then `[]`.

**Comment by lcpriest** [comment]
That's super weird. I'll try recreate it now and fix it. Thanks!

**Comment by lcpriest** [comment]
Fixed by #65
